# error: no matching function for call to 'QComboBox::QComboBox(bool)'
# error: no matching function for call to 'PatternBox::PatternBox(bool)'

--- qtiplot/src/plot2D/dialogs/LayerDialog.cpp.orig	2010-09-27 09:12:48.000000000 +0545
+++ qtiplot/src/plot2D/dialogs/LayerDialog.cpp	2022-08-07 00:46:58.000000000 +0545
@@ -69,14 +69,16 @@
 	QGridLayout *gl2 = new QGridLayout(gb2);
 	gl2->addWidget(new QLabel(tr("Horizontal")), 0, 0);
 
-	alignHorBox = new QComboBox( false );
+	alignHorBox = new QComboBox();
+	alignHorBox->setEditable(false);
 	alignHorBox->insertItem( tr( "Center" ) );
 	alignHorBox->insertItem( tr( "Left" ) );
 	alignHorBox->insertItem( tr( "Right" ) );
 	gl2->addWidget(alignHorBox, 0, 1);
 
 	gl2->addWidget(new QLabel( tr( "Vertical" )), 1, 0 );
-	alignVertBox = new QComboBox( false );
+	alignVertBox = new QComboBox();
+	alignVertBox->setEditable(false);
 	alignVertBox->insertItem( tr( "Center" ) );
 	alignVertBox->insertItem( tr( "Top" ) );
 	alignVertBox->insertItem( tr( "Bottom" ) );


--- qtiplot/src/plot2D/dialogs/PlotDialog.cpp.orig	2011-09-13 13:50:36.000000000 +0545
+++ qtiplot/src/plot2D/dialogs/PlotDialog.cpp	2022-08-07 00:40:39.000000000 +0545
@@ -954,7 +954,8 @@
 	l4->setBuddy(boxFirstColor);
 	gl2->addWidget(l4, 0, 0);
 
-	boxPiePattern = new PatternBox(false);
+	boxPiePattern = new PatternBox();
+	boxPiePattern->setEditable(false);
 	gl2->addWidget(boxPiePattern, 1, 1);
 	gl2->setRowStretch(2, 1);
 
@@ -1337,7 +1338,8 @@
 	hb->addWidget(boxCurveOpacity);
 	gl2->addLayout(hb, 1, 1);
 
-	boxPattern = new PatternBox(false);
+	boxPattern = new PatternBox();
+	boxPattern->setEditable(false);
 	gl2->addWidget(boxPattern, 2, 1);
 	
 	QLabel *l6 = new QLabel("&" + tr("Pattern"));
@@ -2126,17 +2128,20 @@
     QGridLayout *gl3 = new QGridLayout(GroupBoxVectEnd);
     labelXEnd = new QLabel(tr( "X End" ));
     gl3->addWidget(labelXEnd, 0, 0);
-	xEndBox = new QComboBox(false);
+	xEndBox = new QComboBox();
+	xEndBox->setEditable(false);
     gl3->addWidget(xEndBox, 0, 1);
 
 	labelYEnd = new QLabel(tr( "Y End" ));
     gl3->addWidget(labelYEnd, 1, 0);
-	yEndBox = new  QComboBox( false);
+	yEndBox = new QComboBox();
+	yEndBox->setEditable(false);
     gl3->addWidget(yEndBox, 1, 1);
 
 	labelPosition = new QLabel(tr( "Position" ));
     gl3->addWidget(labelPosition, 2, 0);
-	vectPosBox = new  QComboBox( false);
+	vectPosBox = new  QComboBox();
+	vectPosBox->setEditable(false);
 	vectPosBox->addItem(tr("Tail"));
 	vectPosBox->addItem(tr("Middle"));
 	vectPosBox->addItem(tr("Head"));


--- qtiplot/src/table/TableDialog.cpp.orig	2011-08-31 20:11:22.000000000 +0545
+++ qtiplot/src/table/TableDialog.cpp	2022-08-07 00:42:42.000000000 +0545
@@ -127,7 +127,8 @@
     labelFormat = new QLabel(tr( "Format:" ));
  	gl1->addWidget(labelFormat, 2, 0);
 
-    formatBox = new QComboBox(false);
+    formatBox = new QComboBox();
+    formatBox->setEditable(false);
     gl1->addWidget(formatBox, 2, 1);
 
 	labelNumeric = new QLabel(tr( "Precision:" ));


--- qtiplot/src/table/SetColValuesDialog.cpp.orig	2010-09-27 09:12:50.000000000 +0545
+++ qtiplot/src/table/SetColValuesDialog.cpp	2022-08-07 01:29:11.000000000 +0545
@@ -72,12 +72,14 @@
 	hbox1->addWidget(end);
 
 	QGridLayout *gl1 = new QGridLayout();
-	functions = new QComboBox(false);
+	functions = new QComboBox();
+	functions->setEditable(false);
 	functions->addItems(muParserScripting::functionsList(true));
 	gl1->addWidget(functions, 0, 0);
 	btnAddFunction = new QPushButton(tr( "Add function" ));
 	gl1->addWidget(btnAddFunction, 0, 1);
-	boxColumn = new QComboBox(false);
+	boxColumn = new QComboBox();
+	boxColumn->setEditable(false);
 	gl1->addWidget(boxColumn, 1, 0);
 	btnAddCol = new QPushButton(tr( "Add column" ));
 	gl1->addWidget(btnAddCol, 1, 1);


--- qtiplot/src/matrix/MatrixValuesDialog.cpp.orig	2010-09-27 09:12:46.000000000 +0545
+++ qtiplot/src/matrix/MatrixValuesDialog.cpp	2022-08-07 00:41:00.000000000 +0545
@@ -70,7 +70,8 @@
 	endCol->setRange(1, INT_MAX);
 	gl1->addWidget(endCol, 1, 3);
 
-	functions = new QComboBox(false);
+	functions = new QComboBox();
+	functions->setEditable(false);
 	functions->addItems(scriptEnv->mathFunctions());
 	btnAddFunction = new QPushButton(tr( "Add &Function" ));
 	btnAddCell = new QPushButton(tr( "Add Ce&ll" ));
